SSA/Projects/InstCombine/ForLean.lean

```diff
@@ -208,14 +209,33 @@ lemma toNat_neq_of_neq_ofNat {a : BitVec w} {n : Nat} (h : a ≠ n#w) : a.toNat
     rw [Nat.mod_eq_of_lt hn]
   contradiction
 
+lemma neg_neg {a : BitVec w} : - - a = a := by
+  by_cases h : a = 0
+  · subst h
+    simp
+  · rw [toNat_eq]
+    rw [toNat_neg]
+    rw [toNat_neg]
+    have h2 : BitVec.toNat a < 2 ^w := BitVec.toNat_lt a
+    rw [toNat_eq] at h
+    simp at h
+    rw [Nat.mod_eq_of_lt]
+    rw [Nat.mod_eq_of_lt]
+    omega
+    omega
+    rw [Nat.mod_eq_of_lt]
+    omega
+    omega
+
 lemma neg_neq_iff_neq_neg {a b : BitVec w} : -a ≠ b ↔ a ≠ -b:= by
```

Comment:
It's a shame that we need to reprove things like this. This in mathlib I am willing to bet follows from a general theory of [involutive functions](https://en.wikipedia.org/wiki/Involution_(mathematics))